<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from src/robot_description/urdf/robot.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="my_quadruped">
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.4 0.15 0.08"/>
      </geometry>
      <material name="gray">
        <color rgba="0.502 0.502 0.502 1"/>
      </material>
    </visual>
  </link>
  <joint name="lf_abad_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.15 0.1 0"/>
    <parent link="base_link"/>
    <child link="lf_hip_base_link"/>
    <axis xyz="1 0 0"/>
    <limit effort="10.0" lower="-0.7" upper="0.7" velocity="10.0"/>
  </joint>
  <link name="lf_hip_base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.08 0.04 0.06"/>
      </geometry>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="lf_motor_l_joint" type="revolute">
    <origin rpy="0 0 0" xyz="-0.02 0 0"/>
    <parent link="lf_hip_base_link"/>
    <child link="lf_upper_l_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="lf_upper_l_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <geometry>
        <box size="0.02 0.01 0.1"/>
      </geometry>
      <material name="red">
        <color rgba="1.0 0.8 0.796 1"/>
      </material>
    </visual>
  </link>
  <joint name="lf_knee_l_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1"/>
    <parent link="lf_upper_l_link"/>
    <child link="lf_lower_l_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="lf_lower_l_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <box size="0.015 0.008 0.12"/>
      </geometry>
      <material name="blue">
        <color rgba="0.678 0.847 0.902 1"/>
      </material>
    </visual>
  </link>
  <joint name="lf_motor_r_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.02 0 0"/>
    <parent link="lf_hip_base_link"/>
    <child link="lf_upper_r_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="lf_upper_r_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <geometry>
        <box size="0.02 0.01 0.1"/>
      </geometry>
      <material name="green">
        <color rgba="0.565 0.933 0.565 1"/>
      </material>
    </visual>
  </link>
  <joint name="lf_knee_r_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1"/>
    <parent link="lf_upper_r_link"/>
    <child link="lf_lower_r_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="lf_lower_r_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <box size="0.015 0.008 0.12"/>
      </geometry>
      <material name="yellow">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="lf_foot_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.12"/>
    <parent link="lf_lower_l_link"/>
    <child link="lf_foot_link"/>
  </joint>
  <link name="lf_foot_link">
    <visual>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="purple">
        <color rgba="0.502 0.0 0.502 1"/>
      </material>
    </visual>
  </link>
  <joint name="rf_abad_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.15 -0.1 0"/>
    <parent link="base_link"/>
    <child link="rf_hip_base_link"/>
    <axis xyz="1 0 0"/>
    <limit effort="10.0" lower="-0.7" upper="0.7" velocity="10.0"/>
  </joint>
  <link name="rf_hip_base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.08 0.04 0.06"/>
      </geometry>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="rf_motor_l_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.02 0 0"/>
    <parent link="rf_hip_base_link"/>
    <child link="rf_upper_l_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="rf_upper_l_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <geometry>
        <box size="0.02 0.01 0.1"/>
      </geometry>
      <material name="red">
        <color rgba="1.0 0.8 0.796 1"/>
      </material>
    </visual>
  </link>
  <joint name="rf_knee_l_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1"/>
    <parent link="rf_upper_l_link"/>
    <child link="rf_lower_l_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="rf_lower_l_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <box size="0.015 0.008 0.12"/>
      </geometry>
      <material name="blue">
        <color rgba="0.678 0.847 0.902 1"/>
      </material>
    </visual>
  </link>
  <joint name="rf_motor_r_joint" type="revolute">
    <origin rpy="0 0 0" xyz="-0.02 0 0"/>
    <parent link="rf_hip_base_link"/>
    <child link="rf_upper_r_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="rf_upper_r_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <geometry>
        <box size="0.02 0.01 0.1"/>
      </geometry>
      <material name="green">
        <color rgba="0.565 0.933 0.565 1"/>
      </material>
    </visual>
  </link>
  <joint name="rf_knee_r_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1"/>
    <parent link="rf_upper_r_link"/>
    <child link="rf_lower_r_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="rf_lower_r_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <box size="0.015 0.008 0.12"/>
      </geometry>
      <material name="yellow">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="rf_foot_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.12"/>
    <parent link="rf_lower_l_link"/>
    <child link="rf_foot_link"/>
  </joint>
  <link name="rf_foot_link">
    <visual>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="purple">
        <color rgba="0.502 0.0 0.502 1"/>
      </material>
    </visual>
  </link>
  <joint name="lh_abad_joint" type="revolute">
    <origin rpy="0 0 0" xyz="-0.15 0.1 0"/>
    <parent link="base_link"/>
    <child link="lh_hip_base_link"/>
    <axis xyz="1 0 0"/>
    <limit effort="10.0" lower="-0.7" upper="0.7" velocity="10.0"/>
  </joint>
  <link name="lh_hip_base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.08 0.04 0.06"/>
      </geometry>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="lh_motor_l_joint" type="revolute">
    <origin rpy="0 0 0" xyz="-0.02 0 0"/>
    <parent link="lh_hip_base_link"/>
    <child link="lh_upper_l_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="lh_upper_l_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <geometry>
        <box size="0.02 0.01 0.1"/>
      </geometry>
      <material name="red">
        <color rgba="1.0 0.8 0.796 1"/>
      </material>
    </visual>
  </link>
  <joint name="lh_knee_l_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1"/>
    <parent link="lh_upper_l_link"/>
    <child link="lh_lower_l_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="lh_lower_l_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <box size="0.015 0.008 0.12"/>
      </geometry>
      <material name="blue">
        <color rgba="0.678 0.847 0.902 1"/>
      </material>
    </visual>
  </link>
  <joint name="lh_motor_r_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.02 0 0"/>
    <parent link="lh_hip_base_link"/>
    <child link="lh_upper_r_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="lh_upper_r_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <geometry>
        <box size="0.02 0.01 0.1"/>
      </geometry>
      <material name="green">
        <color rgba="0.565 0.933 0.565 1"/>
      </material>
    </visual>
  </link>
  <joint name="lh_knee_r_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1"/>
    <parent link="lh_upper_r_link"/>
    <child link="lh_lower_r_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="lh_lower_r_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <box size="0.015 0.008 0.12"/>
      </geometry>
      <material name="yellow">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="lh_foot_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.12"/>
    <parent link="lh_lower_l_link"/>
    <child link="lh_foot_link"/>
  </joint>
  <link name="lh_foot_link">
    <visual>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="purple">
        <color rgba="0.502 0.0 0.502 1"/>
      </material>
    </visual>
  </link>
  <joint name="rh_abad_joint" type="revolute">
    <origin rpy="0 0 0" xyz="-0.15 -0.1 0"/>
    <parent link="base_link"/>
    <child link="rh_hip_base_link"/>
    <axis xyz="1 0 0"/>
    <limit effort="10.0" lower="-0.7" upper="0.7" velocity="10.0"/>
  </joint>
  <link name="rh_hip_base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.08 0.04 0.06"/>
      </geometry>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="rh_motor_l_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.02 0 0"/>
    <parent link="rh_hip_base_link"/>
    <child link="rh_upper_l_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="rh_upper_l_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <geometry>
        <box size="0.02 0.01 0.1"/>
      </geometry>
      <material name="red">
        <color rgba="1.0 0.8 0.796 1"/>
      </material>
    </visual>
  </link>
  <joint name="rh_knee_l_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1"/>
    <parent link="rh_upper_l_link"/>
    <child link="rh_lower_l_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="rh_lower_l_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <box size="0.015 0.008 0.12"/>
      </geometry>
      <material name="blue">
        <color rgba="0.678 0.847 0.902 1"/>
      </material>
    </visual>
  </link>
  <joint name="rh_motor_r_joint" type="revolute">
    <origin rpy="0 0 0" xyz="-0.02 0 0"/>
    <parent link="rh_hip_base_link"/>
    <child link="rh_upper_r_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="rh_upper_r_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <geometry>
        <box size="0.02 0.01 0.1"/>
      </geometry>
      <material name="green">
        <color rgba="0.565 0.933 0.565 1"/>
      </material>
    </visual>
  </link>
  <joint name="rh_knee_r_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.1"/>
    <parent link="rh_upper_r_link"/>
    <child link="rh_lower_r_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10.0" lower="-3.14" upper="3.14" velocity="10.0"/>
  </joint>
  <link name="rh_lower_r_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <box size="0.015 0.008 0.12"/>
      </geometry>
      <material name="yellow">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="rh_foot_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.12"/>
    <parent link="rh_lower_l_link"/>
    <child link="rh_foot_link"/>
  </joint>
  <link name="rh_foot_link">
    <visual>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="purple">
        <color rgba="0.502 0.0 0.502 1"/>
      </material>
    </visual>
  </link>
</robot>
